src/App.tsx

```diff
@@ -1,14 +1,64 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import cn from 'classnames';
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import { USER_ID, getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+
+enum TodoStatus {
+  Active = 'active',
+  Completed = 'completed',
+  All = 'all',
+}
+
+function getFilteredTodos(todos: Todo[], filter: TodoStatus): Todo[] {
+  switch (filter) {
+    case TodoStatus.Active:
+      return todos.filter(todo => !todo.completed);
+    case TodoStatus.Completed:
+      return todos.filter(todo => todo.completed);
+    case TodoStatus.All:
+    default:
+      return [...todos];
+  }
+}
+
+function getActiveTodosCount(todos: Todo[]): number {
+  return todos.filter(todo => !todo.completed).length;
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState<string>('');
```

Comment:
use enum for error messages
```suggestion
  const [errorMessage, setError] = useState<string>('');
```